spec/features/merchants/items/index_spec.rb

```diff
@@ -13,12 +13,255 @@
+
+              merchant_stephen = Merchant.create!(name: "Stephen's Shady Store")
+              merchant_roger = Merchant.create!(name: "Roger's Fancy Store")
+
+              item_toothpaste = merchant_stephen.items.create!(name: "Item Toothpaste", description: "The worst toothpaste you can find", unit_price: 4000 )
+              item_lamp = merchant_roger.items.create!(name: "Item Lamp", description: "You bet, it's a lamp", unit_price: 4000)
+
+              visit merchant_items_path(merchant_stephen)
+
+              expect(page).to have_link("Create a New Item")
+
+              click_link("Create a New Item")
+
+              expect(current_path).to eq(new_merchant_item_path(merchant_stephen))
+              
+              fill_in "Name", with: "Sword"
+              fill_in "Description", with: "A big sword"
+              fill_in "Unit price", with: "1000"
+              click_button("Submit")
+
+              expect(current_path).to eq(merchant_items_path(merchant_stephen))
+              expect(page).to have_content("Sword")
+              expect(page).to have_link(item_toothpaste.name)
+
+              the_new_item = Item.find_by(name: "Sword")
+              expect(the_new_item.enabled).to eq(false)
+            end
+          end
+        end
+      end
+    end
+  end
+
+  describe 'Merchant Items Index: 5 most popular items' do
+    describe 'When I visit my items index page' do
+      describe 'Then I see the names of the top 5 most popular items ranked by total revenue generated' do
+        describe 'And I see that each item name links to my merchant item show page for that item' do
+          it 'And I see the total revenue generated next to each item name' do
+
+            customer = Customer.create!(first_name: "Kanye", last_name: "Banjo")
+
+            merchant_stephen = Merchant.create!(name: "Stephen's Shady Store")
+            merchant_roger = Merchant.create!(name: "Roger's Fancy Store")
+
+            item_ski = merchant_stephen.items.create!(name: "Ski", description: "fast skis", unit_price: 10) 
+            item_bike = merchant_stephen.items.create!(name: "Bike", description: "danger bike", unit_price: 10)
+            item_climbing_shoes = merchant_stephen.items.create!(name: "Climbing Shoes", description: "fun shoes", unit_price: 10) 
+            item_snowboard = merchant_stephen.items.create!(name: "Snowboard", description: "slow board", unit_price: 10) 
+            item_rock = merchant_stephen.items.create!(name: "Rock", description: "a good rock", unit_price: 10) 
+
+            item_toothpaste = merchant_stephen.items.create!(name: "Toothpaste", description: "The worst toothpaste you can find", unit_price: 10 )
+            item_bowling_shoes = merchant_stephen.items.create!(name: "Bowling Shoes", description: "not fun shoes", unit_price: 10)
+            item_lamp = merchant_roger.items.create!(name: "Item Lamp", description: "You bet, it's a lamp", unit_price: 10)
```

Comment:
This is not actually testing whether the top 5 items is only for one merchant, since the item_lamp is also the 6th-best selling item in the test. So it's not showing up because it's not in the top 5 sellers, not because it doesn't belong to merchant_stephen. See my comment on the top 5 model method for more discussion

Reply:
Good eye.  To correct this I added another transaction, invoice, invoice_item for merchant_roger's item_lamp, with the highest number of sales.  Although this is now the highest selling item, this does not appear in merchant_stephen's top items.